src/encoding/text.rs

```diff
@@ -33,15 +33,98 @@ use std::borrow::Cow;
 use std::collections::HashMap;
 use std::fmt::Write;
 
+/// Functions to encode metrics to the OpenMetrics text format in parts.
+/// Useful when assembling metrics from multiple registries.
+/// Prefer calling [`encode`](encode) when encoding a single registry.
+///
+/// Here is a simple example:
+/// ```
+/// # use prometheus_client::encoding::text::parts;
+/// # use prometheus_client::metrics::counter::Counter;
+/// # use prometheus_client::metrics::histogram::{Histogram, exponential_buckets};
+/// # use prometheus_client::registry::Registry;
+/// #
+/// # fn main() -> Result<(), std::fmt::Error> {
```

Comment:
Aren't examples automatically wrapped in a `main`? Or is this needed for the special return type?

Reply:
It's for the return type.